bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/util/ShellyVersionDTO.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.shelly.internal.util;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * {@link ShellyVersionDTO} compares 2 version strings.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+public class ShellyVersionDTO {
+    private class VersionTokenizer {
+        private final String versionString;
+        private final int length;
+
+        private int position;
+        private int number;
+        private String suffix = "";
+
+        public VersionTokenizer(@Nullable String versionString) {
+            if (versionString == null) {
+                throw new IllegalArgumentException("versionString is null");
+            }
+
+            this.versionString = versionString;
+            length = versionString.length();
+        }
+
+        private boolean MoveNext() {
+            number = 0;
+            suffix = "";
+
+            // No more characters
+            if (position >= length) {
+                return false;
+            }
+
+            while (position < length) {
+                char c = versionString.charAt(position);
+                if (c < '0' || c > '9') {
+                    break;
+                }
+                number = number * 10 + (c - '0');
+                position++;
+            }
+
+            int suffixStart = position;
+
+            while (position < length) {
+                char c = versionString.charAt(position);
+                if (c == '.') {
+                    break;
+                }
+                position++;
+            }
+
+            suffix = versionString.substring(suffixStart, position);
+
+            if (position < length) {
+                position++;
+            }
+
+            return true;
+        }
+
+        private int getNumber() {
+            return number;
+        }
+
+        private String getSuffix() {
+            return suffix;
+        }
+    }
+
+    public boolean equals(String o1, String o2) {
+        return compare(o1, o2) == 0;
+    }
+
+    public int compare(Object o1, Object o2) {
+        String version1 = (String) o1;
+        String version2 = (String) o2;
```

Comment:
Why couldn't you just do this?
```suggestion
    public int compare(String version1, String version2) {
```

Reply:
done

Reply:
changed

Reply:
changed